### PR TITLE
[Feature] Suport CDC job definition persistence

### PR DESCRIPTION
**Pull request by hdygxsj**
### Purpose

Completed CRUD for CDC job definition
截屏2023-11-27 22 28 53
截屏2023-11-27 22 28 58

### CODE REVIEW

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/mapper/CdcJobDefinitionMapper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+package org.apache.paimon.web.server.mapper;
+
+import org.apache.paimon.web.server.data.model.CdcJobDefinition;
+
+import com.baomidou.mybatisplus.core.mapper.BaseMapper;
+
+/**
+ * @author zhongyangyang
+ * @description 针对表【cdc_job_definition】的数据库操作Mapper
```
Comment (s7monk) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/mapper/CdcJobDefinitionMapper.java:27]: Please use English comments and do not need to include author information.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/CdcJobDefinitionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import org.apache.paimon.web.server.data.result.enums.Status;
+import org.apache.paimon.web.server.mapper.CdcJobDefinitionMapper;
+import org.apache.paimon.web.server.service.CdcJobDefinitionService;
+
+import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
+import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
+import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
+import org.springframework.stereotype.Service;
+
+/** @author zhongyangyang */
```
Comment (s7monk) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/CdcJobDefinitionServiceImpl.java:34]: ditto

**Review by s7monk — CHANGES_REQUESTED** [review]

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CdcJobDefinitionDTO.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Data;
+
+@Data
+public class CdcJobDefinitionDTO {
```
Comment (s7monk) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CdcJobDefinitionDTO.java:24]: Please add java doc.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/model/CdcJobDefinition.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import com.baomidou.mybatisplus.annotation.TableName;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import java.io.Serializable;
+
+/** @TableName cdc_job_definition */
```
Comment (s7monk) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/model/CdcJobDefinition.java:31]: CdcJobDefinition entity class.

Comment (hdygxsj) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/data/model/CdcJobDefinition.java]: done

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CdcJobDefinitionDTO.java (hunk 3 above)
Comment (hdygxsj) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CdcJobDefinitionDTO.java:25]: done

**Comment by s7monk** [comment]
LGTM, @SteNicholas @JingsongLi @labbomb PTAL.

File: paimon-web-ui-new/src/api/models/cdc/index.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License. */
+
+
+import type { CdcJobDefinition } from './interface'
+import httpRequest from '@/api/request'
+export * from './interface'
+
+
```
Comment (labbomb) [inline paimon-web-ui-new/src/api/models/cdc/index.ts:23]: Please add a description of each interface, refer to catlog

File: paimon-web-ui-new/src/form-lib/cdc/use-paimon.ts [hunk 6]
```diff
@@ -15,83 +15,50 @@ KIND, either express or implied.  See the License for the
...
-  console.log('item', item.data)
-
+export function usePaimon(item: any) {
+	const { t } = useLocaleHooks()
+	const tabType = item.data.tabType
+	let catalogOptions: any = ref([{value:1}])
+	getAllCatalogs().then((res: any) => {
+		catalogOptions.value = res.data.map((e: any) => { return { value: e.id, label: e.catalogName } })
+	})
+	console.log('item', item.data)
```
Comment (labbomb) [inline paimon-web-ui-new/src/form-lib/cdc/use-paimon.ts:28]: Delete this line.

**Review by labbomb — COMMENTED** [review]
I left some comments, PTAL.

File: paimon-web-ui-new/src/views/cdc/index.tsx [hunk 7]
```diff
@@ -21,12 +21,12 @@ import styles from './index.module.scss';
...
 import { useCDCStore } from '@/store/cdc';
 import type { Router } from 'vue-router';
-
+import { useMessage } from 'naive-ui'
 export default defineComponent({
   name: 'CDCPage',
   setup() {
     const { t } = useLocaleHooks()
-
+    window.$message = useMessage()
```
Comment (labbomb) [inline paimon-web-ui-new/src/views/cdc/index.tsx:29]: Why mount useMessage in window

File: paimon-web-ui-new/src/views/cdc/components/list/index.tsx [hunk 8]
```diff
@@ -247,12 +35,24 @@ export default defineComponent({
...
+                v-model:page-size={this.pagination.pageSize}
+                item-count={this.pagination.count}
+                show-size-picker
+                page-sizes={this.pagination.pageSizes}
+                show-quick-jumper
+                onUpdatePage={this.pagination.onChange}
+                onUpdatePageSize={this.pagination.onUpdatePageSize}
+              />
+            </NSpace>
+          </NSpace>
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/views/cdc/components/list/index.tsx:55]: These components can be automatically imported by `n-space` and do not need to be imported by yourself.

File: paimon-web-ui-new/src/views/cdc/components/list/index.tsx [hunk 9]
```diff
@@ -15,229 +15,17 @@ KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License. */
 
+import { NDataTable, NPagination, NSpace } from 'naive-ui';
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/views/cdc/components/list/index.tsx:18]: remove import

File: paimon-web-ui-new/src/views/cdc/index.tsx (hunk 7 above)
Comment (hdygxsj) [reply paimon-web-ui-new/src/views/cdc/index.tsx:29]: call useMessage() in request.ts will get undefined

File: paimon-web-ui-new/src/views/metadata/components/menu-tree/index.tsx [hunk 10]
```diff
@@ -120,7 +120,7 @@ export default defineComponent({
             <n-space justify='space-between' align='enter'>
               <article>Catalog</article>
               <n-button size='small' quaternary circle>
-                <n-icon>
+                <n-icon >
```
Comment (zhangmo8) [inline paimon-web-ui-new/src/views/metadata/components/menu-tree/index.tsx:123]: What's this ?

Comment (hdygxsj) [reply paimon-web-ui-new/src/views/metadata/components/menu-tree/index.tsx]: I added it by accident. It's been deleted

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**